frame.go

```diff
@@ -87,6 +89,35 @@ func (f *Frame) DataBytes() ([]byte, error) {
 
 }
 
+// SetDataBytes fills the frame data with given raw data. The data must match the pixel format and the dimensions of the frame.
+func (f *Frame) SetDataBytes(data []byte) (err error) {
+
+	// Get the expected buffer length and check data. If data does not have the correct length, we stop and return an
+	// error. Otherwise, pointers will be used to access unexpected memory addresses.
+	expectedBufferLength := C.av_image_get_buffer_size(C.enum_AVPixelFormat(f.c.format), f.c.width, f.c.height, 1)
+	if expectedBufferLength < 0 {
+		return newError(expectedBufferLength)
+	}
+	if int(expectedBufferLength) != len(data) {
+		return fmt.Errorf("astiav: invalid data length of %d for frame. Expected %d", len(data), int(expectedBufferLength))
+	}
+
+	// copy data to c
```

Comment:
2 questions:

1) Why do you need to copy the buffer?
2) If you really need to copy the buffer, could you use `bytesToC()` instead ?

Reply:
1. the C code sets pointers to this buffer, so i perform a copy to avoid accidental  corruption
2. is bytesToC "safe" concerning the garbage collection? the slice must not be collected after the callback is left. Then we could use it...

Reply:
My C knowledge is very limited, but when I read the code of `av_image_fill_arrays()` which uses `av_image_fill_pointers()` correctly, these functions don't copy the data referenced by the pointer. My conclusion: the memory range of the buffer must be copied to a C managed memory area...

Reply:
Thing is, the way I understand it (and I may be wrong), in case of success in your implementation, buffer will never be freed and this will trigger a memory leak (buffer is not freed in C by ffmpeg).

On the other hand, with the implementation I propose, chances are GO GC won't free the buffer until C is not using it anymore. But I'm not 100% sure. I guess this should be tested thoroughly 🤔

Reply:
The more I think about it, `SetDataBytes()` might not be a good idea:

1. Memory management is a problem (I thought that `AVFrame().Free()` would free the buffer, but that is not true)
2. Padding is another problem: data has to be padded with extra bytes depending on resolution and Pixel format (which is not done by `av_image_fill_arrays()`
3. there might be more issues concerning internals of AVFrame

All in all, this seams not to be a good way to load image data to a video frame.

Reply:
> Memory management is a problem (I thought that AVFrame().Free() would free the buffer, but that is not true)

Yeah it seems buffer has to be freed separately using `av_free` 🤔 

> Padding is another problem: data has to be padded with extra bytes depending on resolution and Pixel format

Couldn't this be solved by using `av_image_alloc` before using this function?

Nevertheless I tend to agree with you. What do you think about exporting each individual libav functions (for `DataBytes()` as well) ? For instance there would be `(f *Frame) ImageBufferSize()`, `(f *Frame) CopyImageToBuffer()`, `(f *Frame) AllocImage()`, `(f *Frame) FillImageArrays()`, etc.